### PR TITLE
MRMFeatureFilter support for component group new QCs

### PR DESCRIPTION
**Pull request by pcolaianni**
I decided to give default values to the QCs struct members. I think it's safer than risk uninitialized values.

### CODE REVIEW

File: src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMFeatureQC.h [hunk 1]
```diff
@@ -86,17 +86,17 @@ namespace OpenMS
 
       // Feature members
       /// retention time lower bound
-      double retention_time_l;
+      double retention_time_l { 0.0 };
```
Comment (dmccloskey) [inline src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMFeatureQC.h:89]: The addition of default values is a good idea.  We just want to be careful that in "filter" mode these default values to not lead to the exclusion of peaks without the user being aware.

File: src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMFeatureQC.h [hunk 2]
```diff
@@ -86,17 +86,17 @@ namespace OpenMS
...
       /// retention time upper bound
-      double retention_time_u;
+      double retention_time_u { 100.0 };
```
Comment (dmccloskey) [inline src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMFeatureQC.h]: Let's change this to 1e12 with the above in mind.  This is different than my previous recommendation, but probably better than my previous as well.

File: src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMFeatureQC.h [hunk 3]
```diff
@@ -115,39 +115,39 @@ namespace OpenMS
       String component_group_name;
 
       /// retention time lower bound
-      double retention_time_l;
+      double retention_time_l { 0.0 };
       /// retention time upper bound
-      double retention_time_u;
+      double retention_time_u { 100.0 };
```
Comment (dmccloskey) [inline src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMFeatureQC.h]: same  as above

File: src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFilter.cpp [hunk 4]
```diff
@@ -121,6 +121,32 @@ namespace OpenMS
         {
           if (filter_criteria.component_group_qcs[cg_qc_it].component_group_name == component_group_name)
           {
+            const double rt = features[feature_it].getRT(); //check!
```
Comment (dmccloskey) [inline src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFilter.cpp]: the `//check!` can be removed.  That was a note to myself previously.

File: src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFilter.cpp [hunk 5]
```diff
@@ -197,6 +222,17 @@ namespace OpenMS
...
+            for (const std::pair<String,std::pair<double,double>>& kv : filter_criteria.component_group_qcs[cg_qc_it].meta_value_qc)
+            {
+              bool metavalue_exists {false};
+              if (!checkMetaValue(features[feature_it], kv.first, kv.second.first, kv.second.second, metavalue_exists))
+              {
+                cg_qc_pass = false;
+                cg_qc_fail_message_vec.push_back("metaValue[" + kv.first + "]");
+              }
+              if (metavalue_exists) ++cg_tests_count;
+            }
```
Comment (dmccloskey) [inline src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFilter.cpp:235]: Nice.

**Review by dmccloskey — APPROVED** [review]
Looks good.

**Comment by pcolaianni** [comment]
@dmccloskey PR updated. I also changed the values in _MRMFeatureQCFile.cpp_ for `retention_time_u`

**Review by dmccloskey — APPROVED** [review]
Looks good.

**Pull request closed without merging**